src/portable/nordic/nrf5x/dcd_nrf5x.c

```diff
@@ -57,6 +56,13 @@
 #include "mcu/mcu.h"
 #endif
 
+#if 1000*MDK_MAJOR_VERSION + MDK_MINOR_VERSION <= 8040
+// Nordic actually has generated a mess here: nrfx==1.9.0 has MDK 8.40.3 while nrfx==2.0.0 has MDK 8.29.0
```

Comment:
They concurrently developed both 1.x and 2.x together since it is breaking changes. 1.9 is actually released after 2.0. It is supprising that nordicsemi didn't include NRFX_VERSION in their codebase. 

This check make version v2.0 to v2.4 not compilable with tinyusb which is not correct. The correct check must ensure:
- all v2.x compile as it is
- additionally some of 1.x can be compiled, not all since there is a slash of MDK version.

From what I found MDK version for releases are

```
- v2.6.0: 8.44.1
- v2.5.0: 8.40.2
- v2.4.0: 8.37.0
- v2.3.0: 8.35.0 
- v2.2.0: 8.32.1
- v2.1.0: 8.30.2
- v2.0.0: 8.29.0

- v1.9.0: 8.40.3
- v1.8.6: 8.35.0 <- conflict with 2.3.0
- v1.8.5: 8.32.3
- v1.8.4: 8.32.1 <- conflict with 2.2.0
- v1.8.2: 8.32.1 <- conflict with 2.2.0
- v1.8.1: 8.27.1
```

Therefore the correct check would be:
- MDK < 8.29.0 (v2.0), MDK != 8.40.3, 32.3, 27.1

v1.x conflict with v1.x is not supported since there is no way to tell the difference. User of those version must upgrade to other 1.x version. 

PS: Do we need to support all 1.x version, if not, we can just only inlcude the 1.9.0 using its MDK version. It is realative easy to upgrade to 1.9

Reply:
I've changed the code so that the old API is correctly detected.  But, I put priority on detection of the old API in case of conflicts, because TMO most people now keep the versions current because handling of the Connect SDK is different.

Upgrading to nrfx 1.9.0 is not that easy, because the nrfx is part of the Nordic SDK.